caas/kubernetes/provider/annotations.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package provider
+
+import (
+	"github.com/juju/juju/core/annotations"
+)
+
+const (
+	// AnnotationJujuStorageName is the Juju annotation that represents a
+	// storage objects associated Juju name.
+	AnnotationJujuStorageName = "storage.juju.is/name"
+
+	// AnnotationJujuVersion is the version annotation used on operator
+	// deployments.
+	AnnotationJujuVersion = "juju.is/version"
+
+	// legacyAnnotationStorageName is the legacy annotation used by Juju for
+	// dictating storage name on k8s storage objects.
+	legacyAnnotationStorageName = "juju-storage"
+
+	// legacyAnnotationVersion is the legacy annotation used by Juju for
+	// dictating juju agent version on operators.
+	legacyAnnotationVersion = "juju-version"
+)
+
+type AnnotationKeySupplier func() string
+
+// AnnotationsForStorage provides the annotations that should be placed on a
+// storage object. The annotations returned by this function are storage
+// specific only and should be combined with other annotations where
+// appropriate.
+func AnnotationsForStorage(name string, legacy bool) annotations.Annotation {
```

Comment:
Same with these - they are only used in the provider package

Reply:
we could make `AnnotationsForStorage` and `AnnotationsForVersion` consistent